feat: implement review creation API

moominIsCute opened the pull request:
## #️⃣ Issue Number<!--- ex) #이슈번호, #이슈번호 -->

- Closes #11 

## 📝 요약(Summary)<!--- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->
- Review 엔티티 equipmentScore 제거, userScore만 유지
- ReviewCreateRequest, ReviewResponse DTO 생성 (Record 방식)
- ReviewCommandService 및 Impl 구현
- ReviewController 생성 (POST /reviews)
- 사용자 점수 1~5점 검증 추가
- User 존재 여부 확인 로직 추가

## 🛠️ PR 유형
어떤 변경 사항이 있나요?
- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## ✅ PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.
- [x] 커밋 메시지 컨벤션에 맞게 작성

moominIsCute committed:
feat(review): implement review creation API

- Review 엔티티 equipmentScore 제거, userScore만 유지
- ReviewCreateRequest, ReviewResponse DTO 생성 (Record 방식)
- ReviewCommandService 및 Impl 구현
- ReviewController 생성 (POST /reviews)
- 사용자 점수 1~5점 검증 추가
- User 존재 여부 확인 로직 추가

moominIsCute commented on src/main/java/com/golfRental/domain/review/controller/ReviewController.java:
```diff
@@ -0,0 +1,31 @@
+    private final ReviewCommandService reviewCommandService;
+
+    //리뷰 생성
+    @PostMapping("/{reservationId}")
```
예약 도메인과 논의가 필요한 부분입니다. 현재 mvp수준으로 12월 1일 이전에 코드개선이 이루어 질 것입니다.

moominIsCute commented on src/main/java/com/golfRental/domain/review/dto/response/ReviewResponse.java:
```diff
@@ -0,0 +1,27 @@
+    public static ReviewResponse from(Review review) {
+        return new ReviewResponse(
+                review.getId(),
+                review.getUser().getId(),
+                review.getUser().getName(),
```
목록조회시 참고 하겠습니다.

moominIsCute committed:
refactor(review): separate controller into command and query

- ReviewController 구현체 생성

moominIsCute committed:
Merge remote-tracking branch 'origin/dev' into feat/11-review

moominIsCute committed:
Merge remote-tracking branch 'origin/dev' into feat/11-review

moominIsCute committed:
feat(review): implement review creation API

- 리뷰 생성 시 User 존재 여부 검증 로직 추가

moominIsCute commented on src/main/java/com/golfRental/domain/review/controller/ReviewController.java:
```diff
@@ -0,0 +1,31 @@
+    @PostMapping("/{reservationId}")
+    public ResponseEntity<CommonApiResponse<ReviewResponse>> createReview(
+            @Valid @RequestBody ReviewCreateRequest request
+    ) {
```
reservation 도메인 생성시 협업해야할 사항

AllSungho commented on src/main/java/com/golfRental/domain/review/service/command/ReviewCommandServiceImpl.java:
```diff
@@ -0,0 +1,36 @@
+public class ReviewCommandServiceImpl implements ReviewCommandService {
+
+    private final ReviewRepository reviewRepository;
+    private final UserRepository userRepository;
```
UserRepository를 직접 참조를 하는데, 이는 UserRepo와 ReviewCommandService간 결합도가 높아지므로,
저희는 Sevice인터페이스를 참조하기로 했습니다!
따라서, UserRepository가 아닌 UserCommandService 또는 UserQueryService를 참조하는 것이 역할과 책임의 경계 명확화하는데에 좋을 것으로 보입니다!

moominIsCute replied:
반영했습니다

moominIsCute committed:
refactor(review): extract userId from SecurityContext instead of request body

- ReviewCreateRequest에서 userId 필드 제거
- Controller에서 @AuthenticationPrincipal AuthUser로 userId 획득
- ReviewCommandService 시그니처 변경: createReview(Long userId, request)
- UserRepository 직접 참조 → UserQueryService로 변경하여 결합도 감소
